src/autopas/utils/RaplMeter.h

```diff
@@ -0,0 +1,72 @@
+/**
+ * @file RaplMeter.h
+ * @author Vincent Fischer
+ * @date 09.11.2021
+ */
+
+#pragma once
+
+#include <cstdint>
+#include <cstdio>
+#include <fstream>
+
+namespace autopas::utils {
+
+/**
+ * Measures the energy consumption between calls of reset() and sample()
+ * using perf events. This requires root or a paranoid value less than 1.
+ */
+class RaplMeter {
+ private:
+  int _type;
+  long _psys_raw, _pkg_raw, _cores_raw, _ram_raw;
+  double _psys_unit, _pkg_unit, _cores_unit, _ram_unit;
+  int _psys_config, _pkg_config, _cores_config, _ram_config;
+  int _psys_fd, _pkg_fd, _cores_fd, _ram_fd;
+
+  void open_msr();
+  uint64_t read_msr(int msr_offset);
+  int open_perf_event(int type, int config);
+  long read_perf_event(int fd);
+
+ public:
+  ~RaplMeter();
+  /** initialisation may fail, so moved out of constructor */
+  void init();
+  /** reset perf file descriptors to start new measurement */
+  void reset();
+  /** measure power consumption since last call of reset
+   * the results can be retrieved with the get_<domain>_energy() functions.
+   */
+  void sample();
```

Comment:
```suggestion
  /** 
   * Reset perf file descriptors to start a new measurement.
   */
  void reset();
  /** 
   * Measure power consumption since the last call to reset.
   * The results can be retrieved with the get_<domain>_energy() functions.
   */
  void sample();
```

Just doc style. Please adapt all others as well.